Source/WebCore/css/parser/CSSPropertyParserConsumer+Anchor.cpp

```diff
@@ -0,0 +1,216 @@
+    case CSSValueSpanSelfEnd:
+        return Type::SelfAmbiguous;
+
+    case CSSValueCenter:
+    case CSSValueSpanAll:
+        return Type::Axesless;
+
+    default:
+        return { };
+    }
+}
+
+static bool typesAreCompatible(Type type1, Type type2)
+{
+    switch (type1) {
+    case Type::PhysicalX:
+        return (type2 == Type::PhysicalY || type2 == Type::Axesless);
+
+    case Type::PhysicalY:
+        return (type2 == Type::PhysicalX || type2 == Type::Axesless);
+
+    case Type::LogicalBlock:
+        return (type2 == Type::LogicalInline || type2 == Type::Axesless);
+
+    case Type::LogicalInline:
+        return (type2 == Type::LogicalBlock || type2 == Type::Axesless);
+
+    case Type::SelfLogicalBlock:
+        return (type2 == Type::SelfLogicalInline || type2 == Type::Axesless);
+
+    case Type::SelfLogicalInline:
+        return (type2 == Type::SelfLogicalBlock || type2 == Type::Axesless);
+
+    case Type::Ambiguous:
+        return (type2 == Type::Ambiguous || type2 == Type::Axesless);
+
+    case Type::SelfAmbiguous:
+        return (type2 == Type::SelfAmbiguous || type2 == Type::Axesless);
+
+    case Type::Axesless:
+        return true;
+    }
+
+    ASSERT_NOT_REACHED();
+    return false;
+}
+
+static bool typeIsAxisUnambiguous(Type type)
+{
+    switch (type) {
+    case Type::Ambiguous:
+    case Type::SelfAmbiguous:
+    case Type::Axesless:
+        return false;
+
+    default:
+        return true;
+    }
+}
+
+RefPtr<CSSValue> consumePositionArea(CSSParserTokenRange& range, const CSSParserContext&)
+{
+    auto dim1 = consumeIdent(range);
+    if (!dim1)
+        return nullptr;
+    if (dim1->valueID() == CSSValueNone)
+        return dim1;
+
+    auto maybeDim1Type = determineType(dim1->valueID());
+    if (!maybeDim1Type)
+        return nullptr;
+    auto dim1Type = *maybeDim1Type;
+
+    auto dim2 = consumeIdent(range);
+    if (!dim2)
+        return dim1;
+
+    auto maybeDim2Type = determineType(dim2->valueID());
+    if (!maybeDim2Type)
+        return nullptr;
+    auto dim2Type = *maybeDim2Type;
+
+    if (!typesAreCompatible(dim1Type, dim2Type))
+        return nullptr;
+
+    // An keyword "<kw>" that's unambiguous about its axis is equivalent to "<kw> span-all" or "span-all <kw>"
+    // If this is the case, don't include the span-all.
+    if (dim1->valueID() == CSSValueSpanAll && typeIsAxisUnambiguous(dim2Type))
+        return dim2;
+    if (typeIsAxisUnambiguous(dim1Type) && dim2->valueID() == CSSValueSpanAll)
+        return dim1;
+
+    // Ensure the X/block axis keyword goes first in the pair.
+    if (dim1Type == Type::PhysicalY || dim1Type == Type::LogicalInline || dim1Type == Type::SelfLogicalInline || dim2Type == Type::PhysicalX || dim2Type == Type::LogicalBlock || dim2Type == Type::SelfLogicalBlock)
+        return CSSValuePair::create(dim2.releaseNonNull(), dim1.releaseNonNull());
+
+    return CSSValuePair::create(dim1.releaseNonNull(), dim2.releaseNonNull());
```

Comment:
The job of the parser is to do the parsing, so how about we make this the _only_ place that we handle this ordering and optionality logic?

Make this function _always_ return two values (unless it's `none`), with a guaranteed axis ordering for a given type of axes.